storage_controller/src/service.rs

```diff
@@ -2082,6 +2083,40 @@ impl Service {
             }
         }
 
+        let preferred_azs = {
+            let locked = self.inner.read().unwrap();
+            response_shards
+                .iter()
+                .filter_map(|resp| {
+                    let az_id = locked
+                        .nodes
+                        .get(&resp.node_id)
+                        .map(|n| n.get_availability_zone_id().to_string())?;
+
+                    Some((resp.shard_id, az_id))
+                })
+                .collect::<Vec<_>>()
+        };
+
+        let updated = self
```

Comment:
It feels a bit strange to go back and update the rows we only just created.  Can we pick the home AZ at the start of tenant creation instead?  One can "peek" at which AZ the scheduler would select by just doing a dummy schedule call.

Reply:
Peeking is a nice idea. I'll try it out.

Reply:
https://github.com/neondatabase/neon/pull/8937/commits/6a3ce827e816bf56746fbb2e568b51202731e939